bloock-bridge/src/server/record.rs

```diff
@@ -310,37 +278,33 @@ fn build_record(
                 let encrypter_arguments = match encrypt.args {
                     Some(encrypter_arguments) => encrypter_arguments,
                     None => {
-                        return RecordBuilderResponse {
-                            record: None,
-                            error: Some(Error {
-                                kind: BridgeError::RecordError.to_string(),
-                                message: "no arguments provided".to_string(),
-                            }),
-                        }
+                        return record_builder_response_error("no arguments provided".to_string())
                     }
                 };
-                let secret = match encrypter_arguments.secret {
-                    Some(secret) => secret,
-                    None => {
-                        return RecordBuilderResponse {
-                            record: None,
-                            error: Some(Error {
-                                kind: BridgeError::RecordError.to_string(),
-                                message: "no secret provided".to_string(),
-                            }),
+
+                let key = match encrypter_arguments.key {
+                    Some(key) => key,
+                    None => return record_builder_response_error("no key provided".to_string()),
+                };
+
+                let key = match hex::decode(key) {
```

Comment:
I guess this implies that the key provided by the user is a hex. 

Maybe we should do the opposite. Allow to provide non-hexadecimal values and encode it.

Reply:
By allowing the user to provide non-hex values, what format do you mean they are going to be providing?

The key has to be an array of 32 bytes. I encoded it in hex because I though it's easier for the user to save it like that, than keeping an array of bytes. The key can probably be generated by the generate keys function, and then the user can store it somewhere else.